src/warnet/test_framework_bridge.py

```diff
@@ -302,3 +310,69 @@ def parse_args(self):
                 self.options.descriptors = None
 
         PortSeed.n = self.options.port_seed
+
+    def connect_nodes(self, a, b, *, peer_advertises_v2=None, wait_for_connect: bool = True):
+        """
+        Kwargs:
+            wait_for_connect: if True, block until the nodes are verified as connected. You might
+                want to disable this when using -stopatheight with one of the connected nodes,
+                since there will be a race between the actual connection and performing
+                the assertions before one node shuts down.
+        """
+        from_connection = self.nodes[a]
+        to_connection = self.nodes[b]
```

Comment:
the test framework in warnet also has `self.warnet` so you can get the tanks directly from there, and if there is `Tank.ipv4` or `Tank.get_external_ip()` working then the I think the rest of this will be a lot simpler.